pkg/service/replay/utils.go

```diff
@@ -104,6 +113,460 @@ func (t *requestMockUtil) SimulateRequest(ctx context.Context, _ uint64, tc *mod
 	return nil, nil
 }
 
+func CalculateAndInsertTestCoverage(ctx context.Context, logger *zap.Logger, reportDB ReportDB, testRunID string, language config.Language) {
```

Comment:
All this code coverage logic is too big for `utils`. We should create a separate package for coverage related code.

Reply:
done!